### PR TITLE
[NEW] Accept quoted slash command arguments

### LINKED ISSUE #11469: Slash Commands Don't Handle New Lines
**Is your feature request related to a problem? Please describe.**
The problem is the slash command gets only first line of message as arguments. All other lines are ignored.

**Describe the solution you'd like**
I think that all lines should be parsed as command arguments along with every new line (\n), so in case of command:

```
/examplecommand hello my name is
Peter and I live
in Poland
```

The command arguments should be processed as:

`["hello", " my", "name", "is", "\n", "Peter", "and", "I", "live", "\n", "in", "Poland"]`

But right now command receives only:

`["hello", "my", "name", "is"]`

**Describe alternatives you've considered**
Maybe there should be some command property like "argumentAsText: boolean" which will pass all given text to the command as one argument:

```
export class ExampleCommandCommand implements ISlashCommand {
  public command: string = 'example';
  public argumentAsText: boolean = true;

  ...
  public async executor(context: SlashCommandContext, ....) {
    this.logger.info(context.getArguments()[0]); // hello, my name is\n ...
  }

```
**Additional context**
---none---

### PR DESCRIPTION
**Pull request by Hudell**
This change will make the app engine recognize quoted strings as a single argument for slash commands, even when they have spaces on it.

It will also accept multiple lines of arguments
Closes #11469

### CODE REVIEW

**Comment by kapkoi** [comment]
Pada Sab, 11 Ogo 2018 1:28 AM Pierre H. Lehnen <notifications@github.com>
menulis:

**Comment by kapkoi** [comment]
packages/rocketchat-apps/server/bridges/commands.js
<https://github.com/RocketChat/Rocket.Chat/pull/11744/files#diff-0>

Pada Sab, 11 Ogo 2018 1:31 AM Kap Koi <kapkoi360@gmail.com>menulis:
packages/rocketchat-apps/server/bridges/commands.js
<https://github.com/RocketChat/Rocket.Chat/pull/11744/files#diff-0>

**Comment by Hudell** [comment]
It does now.

**Comment by rodrigok** [comment]
@Hudell can you fix the lint errors?

**Comment by Hudell** [comment]
@rodrigok done.

**Comment by piotrkochan** [comment]
Could you also resolve https://github.com/RocketChat/Rocket.Chat/issues/11469 ?

**Comment by Hudell** [comment]
I'll look into it.

**Comment by piotrkochan** [comment]
any update on this?

**Comment by d-gubert** [comment]
So I've tested it with the following command:

```
/bitbucket-server "test the branch"
"has it corrected" everything now?
```

And in the app's logs it appeared like this:

clipboard_slahscommand_args

I'm assuming this is not the expected behavior? If it is, let me know and I can approve it :)

**Comment by piotrkochan** [comment]
Do it work with multiple lines without qouting ""?

**Comment by d-gubert** [comment]
Currently, it's concatenating new lines in one arg

The following input:
```
/bitbucket-server alo
test
new line
args
```

Resulted in this parsing:

image

**Review by tassoevan — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Review by sampaiodiego — DISMISSED** [review]

**Review by thassiov — APPROVED** [review]

File: apps/meteor/lib/utils/parseParameters.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			// Replace carriage returns inside the quotes with a PLU character
+			.replace(/\r/gm, '\u008C')
+			// Replace tabs inside the quotes with a VTS character
+			.replace(/\t/g, '\u008A');
+
+		line = line.replace(item, newItem);
+	});
+
+	// If two quoted parameters are not separated by a space, add one automatically
+	line = line.replace(/\u0087\u0086/g, ' ');
```
Comment (pierre-lehnen-rc) [inline apps/meteor/lib/utils/parseParameters.ts:45]:
```suggestion
	line = line.replace(/\u0087\u0086/g, '\u0087 \u0086');
```

this doesn't change the function result but is the right way to do it.

**Review by pierre-lehnen-rc — APPROVED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**